src/client/dfs/SConscript

```diff
@@ -54,19 +56,45 @@ def scons():
                         instver = cache["lustre-client"].installed
             if gotpkg is True:
                 if parse_version(instver.version) >= parse_version('2.12.57'):
+                    """ Lustre version is compatible """
                     gotversion = True
         if gotdistro is False:
             print("Unable to identify distro.")
+            raise ImportError('fallback')
         else:
             if gotpkg is False:
                 print("Unable to find lustre nor lustre-client package.")
+                raise ImportError('fallback')
             else:
                 if gotversion is False:
-                    print("Unable to find minimum version of lustre nor lustre-client package.")
+                    print("Installed Lustre version is not compatible.")
                 else:
-                    denv.AppendUnique(CCFLAGS=['-DLUSTRE_INCLUDE'])
-    except ImportError:
-        raise
+                    print("Installed Lustre version is compatible.")
+    except ImportError as error:
+        print(error.message +
+              ", trying to find Lustre headers in usual locations.")
+        """ Try the empirical way, by just trying to check both required header files are present ... """
+        if path.exists("/usr/include/lustre/lustreapi.h"):
```

Comment:
This is the **only** kind of test that I think this code should be doing IMHO.  It doesn't matter how (RPM, deb, `make install`, etc.) the header files are present, just that they are.  There should be an option to scons to also tell it where the headers are in case they are in a non-standard location.

But frankly, the presence of header files is not usually enough.  You need to know that they represent a compatible API.  For that you typically do API feature compile tests and `scons` provides tools to do that kind of checking without having to write our own python code to do it.

This code should be using those features and only those features for determining the _lustre-client_ compatibility.  Doing that would make this [whole section of code](https://github.com/daos-stack/daos/blob/a3d74025fc61315b1aeffbeb790c98cbd6fcc2a4/src/client/dfs/SConscript#L12-L105) much easier to read, understand and maintain.

Reply:
> scons provides tools to do that kind of checking without having to write our own python code to do it.

Can you be more detailed, or may be point to some existing in our build ?

Reply:
There is a whole [manual for scons](https://scons.org/doc/0.97/HTML/scons-user/book1.html) that details all of the things it can do like [header checks](https://scons.org/doc/0.97/HTML/scons-user/x2827.html), [function availability](https://scons.org/doc/0.97/HTML/scons-user/x2836.html), and the ability to [add custom checks](https://scons.org/doc/0.97/HTML/scons-user/x2867.html).  Among many other things.

These are the tools of scons for determining the capabilities of APIs, without having to write a whole bunch of python code that will need ongoing maintenance.